### DISCUSSION TITLE
Optional params + route groups + route priority

### DISCUSSION THREAD

**Discussion by yasserlens** [sveltejs/kit#7357] [Q&A]
Hello!

I have the following file structure:
routes/[[locale]]/(main)/+page.svelte
routes/[[locale]]/(main)/design/+page.svelte

As far as I understand from the sorting section in the docs ([here](https://kit.svelte.dev/docs/advanced-routing#sorting)), going to:
example.com/design 

should return this page, since "design" is in the path:
routes/[[locale]]/(main)/design/+page.svelte

Instead, it's returning this
routes/[[locale]]/(main)/+page.svelte
Which would consider "design" as the "locale".

Is this working as intended?
If so, how would you get around that?

Thanks!

**Comment by dominikg** [comment]
This could be a bug. Do you have a repository where this is reproducible? 
To work around it you could add a matcher `routes/[[locale=locale]]/...` that validates the first segment must be an existing locale that is supported by your application (advisable anyways, otherwise you end up with /does-not-exist being matched as locale too).

Does it also happen when you move the (main) group in front of the optional param? `routes/(main)/[[locale]]` or without the group? Or is it only happening in this particular order?

**Reply by yasserlens** [reply]
Thanks for the tip, @dominikg  - using a matcher will help with validation. However, this doesn't solve the reported issue.
Now I have a new issue surfaced by the matcher. More on that below.

src/params/matcher.ts
```
export function match(param: string): boolean {
  // matches e.g. en-es string structure.
  const re = new RegExp('([A-Za-z]){2}(-)([A-Za-z]){2}');
  return re.test(param) || param.toLowerCase() === 'global';
}
```

Now, the root route doesn't match anything. I need to specify a locale to get the root route (i.e. the locale isn't optional).

This seems like a bug to me now after testing with more scenarios, see below:

The directory structure is as follows:
routes/
    test/
        [optional]
            (main)
                coming-soon
                   +page.svelte
                +page.svelte
            no-group
              +page.svelte
            +layout.svelte

I placed 2 directories under the [optional] directory:
(main)
and 
no-group

The following visits return the following pages:
http://localhost:5173/test ---> routes/test/[[optional]]/(main)/+page.svelte
http://localhost:5173/test/coming-soon ---> routes/test/[[optional]]/(main)/+page.svelte

**The last visit should have returned** 
routes/test/[[optional]]/(main)/coming-soon/+page.svelte

I tested the other branch, which doesn't have (main) group:
http://localhost:5173/test/no-group
This still returned
routes/test/[[optional]]/(main)/+page.svelte

It should have returned
routes/test/[[optional]]/no-group/+page.svelte

I'm happy to provide more details if needed.

Thanks for looking into this!

**Reply by dominikg** [reply]
Please file an issue and include all information asked for in the form, including a link to a repository with a reproduction https://github.com/sveltejs/kit/issues/new?assignees=&labels=&template=bug_report.yml
